optaplanner-examples/src/main/java/org/optaplanner/examples/common/app/CommonApp.java

```diff
@@ -151,4 +151,14 @@ protected AbstractSolutionExporter createSolutionExporter() {
         return null;
     }
 
+    public Solution_ loadUnsolvedProblemFromResource(String problemFileName) {
```

Comment:
I'd rather not pollute the examples' main code with test related code. I recommend to move this to the test class. See also `TspApp.createSolver()`.

@rsynek what's you opinion?

Reply:
Agreed, these 2 methods belong in PlannerTestUtils.

You can use a generic _method_ argument if you want to avoid the solution cast:

    public <Solution_> Solution_ loadUnsolvedProblemFromResource(Class<Solution_> solutionClass, String problemFileName) {

Reply:
@ge0ffrey The second part of your comment is confusing. I don't see any cast taking place.

Reply:
@yurloc @ge0ffrey the idea was to keep these methods in the *App structure as it already offers means to load datasets, making it possible to use these methods by other code than tests, too.

Reply:
@rsynek I'd keep the responsibilities of the class as limited as possible. Let's avoid adding more features to the class just because there's no better place for them yet and the class already does *something similar*. That's not a good reason. `CommonApp` is a base class for all example apps and shouldn't be misused for creating solvers and loading solutions in tests. In short, let's follow SRP.

Reply:
@yurloc you're right; the CommonApp is not a good place for such methods, placing them there is just a shortcut. Although the PlannerTestUtils sounds like a good option, any utility class violates SRP from it's essence. The hierarchy of *App classes offer a path to directory where datasets are stored, classes responsible for loading the datasets, etc. for every example.

The best option I see is a new class responsible for loading datasets from multiple locations (classpath, data directory) that depends on *App class.